rust/crd/src/s3logdir.rs

```diff
@@ -0,0 +1,170 @@
+    prefix: String,
+}
+
+impl S3LogDir {
+    pub async fn resolve(
+        log_file_dir: Option<&LogFileDirectorySpec>,
+        namespace: Option<String>,
+        client: &stackable_operator::client::Client,
+    ) -> Result<Option<S3LogDir>, Error> {
+        #[allow(irrefutable_let_patterns)]
+        let (s3bucket, prefix) =
+            if let Some(S3(S3LogFileDirectorySpec { bucket, prefix })) = log_file_dir {
+                (
+                    bucket
+                        .resolve(client, namespace.unwrap().as_str())
+                        .await
+                        .context(S3BucketSnafu)
+                        .ok(),
+                    prefix.clone(),
+                )
+            } else {
+                // !!!!!
+                // Ugliness alert!
+                // No point in trying to resolve the connection anymore since there is no
+                // log_file_dir in the first place.
+                // This can casually happen for Spark applications that don't use a history server
+                // !!!!!
+                return Ok(None);
+            };
+
+        // Check that a bucket name has been defined
+        s3bucket
+            .as_ref()
+            .and_then(|i| i.bucket_name.as_ref())
+            .context(BucketNameMissingSnafu)?;
+
+        if let Some(conn) = s3bucket.as_ref().and_then(|i| i.connection.as_ref()) {
+            if let Some(tls) = &conn.tls {
+                match &tls.verification {
+                    TlsVerification::None {} => return S3TlsNoVerificationNotSupportedSnafu.fail(),
+                    TlsVerification::Server(server_verification) => {
+                        match &server_verification.ca_cert {
+                            CaCert::WebPki {} => {}
+                            CaCert::SecretClass(_) => {
+                                return S3TlsCaVerificationNotSupportedSnafu.fail()
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        if let Some(conn) = s3bucket.as_ref().and_then(|i| i.connection.as_ref()) {
+            if conn.tls.as_ref().is_some() {
+                tracing::warn!("The resource indicates S3-access should use TLS: TLS-verification has not yet been implemented \
+            but an HTTPS-endpoint will be used!");
+            }
+        }
+        Ok(Some(S3LogDir {
+            bucket: s3bucket.unwrap(),
+            prefix,
+        }))
+    }
+
+    /// Constructs the properties needed for loading event logs from S3.
+    /// These properties are later written in the `HISTORY_CONFIG_FILE_NAME_FULL` file.
+    ///
+    /// The following properties related to credentials are not included:
+    /// * spark.hadoop.fs.s3a.aws.credentials.provider
+    /// * spark.hadoop.fs.s3a.access.key
+    /// * spark.hadoop.fs.s3a.secret.key
+    /// instead, the environment variables AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY are set
```

Comment:
Seems a bit weird to have different flows for spark applications and history server but fine for me ;)